Added filters in upcoming courses page

shailesh1210 opened the pull request:
-Dropdown filters instead of buttons
-Added functionality to all the filters except for 'All Courses' filter. 
-Same filtering method as in training page

shailesh1210 committed:
Added filters in upcoming courses page

palfi-andras reviewed (approved):
Great Job!

keithgutfreund commented:
Shailesh
Great job on the button filters!   It looks great.  Also, really nice reusing Devin’s course list!.  That is awesome.  Also, you can select multiple filters for a Boolean AND function, like City:London AND Trainer: Dave West.  Great work.

There are just a few bugs that I noticed:
a. If you select a month with the CALENDAR button, it doesn’t change the calendar in the display.
b. ALL COURSES filter doesn’t work.
c. No courses are taught by Dave West - we will have to update Devin’s course list to modify a few.  Also, we should create at least 1 course in April so that the UI shows something when it first starts.
d. If you change the Calendar to July and then filter by ALL COURSE TYPE: SCALED AGILE, you only see 1 class, Certified SAFe Scrum Master.  However, there is also a second SAFE course in July, Certified SAFe Government Practitioner which does not show up in the filtered view.  However, it does appear in the unfiltered view for July.
e. If you select CITY: BOSTON and then CITY: LONDON, you get no results.  I think because you may be requiring the city to be both Boston and London.  I think what you should do is when you click on a CITY then you should first reset any existing city filter before assigning a new value.  The same problem with the other filters.  For example, nothing shows up if you select ALL COUNTRIES: USA and then ALL COUNTRIES: UK.

Again, very nice work.

Keith

devin-dumais commented on src/containers/CoursesList/courses.js:
```diff
@@ -14,11 +14,16 @@ export const courses = [{
   effortDuration: "2 days",
   id: 1,
   img: psm,
+  trainer: "Raj Heda",
```
We should avoid adding redundant data to the course objects.  `trainer` can be removed because the same data can be accessed with ` meetTheInstructors.instructorName`.

shailesh1210 committed:
Addressed all the Keith's comments

shailesh1210 commented:
-Added functionality of "All Courses" Filter. Replaced it with a button instead of selection menu. It displays all the courses available in a given month
- Calendar view changes based on the selection from "Calendar filter"
- "If you select CITY: BOSTON and then CITY: LONDON, you get no results." -- This is solved.
-"If you change the Calendar to July and then filter by ALL COURSE TYPE: SCALED AGILE, you only see 1 class, Certified SAFe Scrum Master. However, there is also a second SAFE course in July, Certified SAFe Government Practitioner which does not show up in the filtered view. However, it does appear in the unfiltered view for July." -- This is solved

shailesh1210 committed:
Bug fixes filters

shailesh1210 commented:
-Bug fixes in the filters

keithgutfreund commented:
Nice job!

keithgutfreund reviewed (approved):
Nice job - ready to go

shailesh1210 committed:
Fixed all the bugs. All the filters are working

shailesh1210 commented:
I checked all the filters and everything is working fine.

devin-dumais commented on src/containers/UpcomingCourses/UpcomingCourses.js:
```diff
@@ -29,7 +28,7 @@ const UpcomingCourses = () => {
     const [monthId, setMonthId] = useState(new Date().getMonth());
    
-    const [allCourses, setAllCourses] = useState(courses);
+    let [allCourses, setAllCourses] = useState(courses);
```
This should remain `const`.  It represents the state of this component on this render, which is constant.  When the value of state changes, the component re-renders.

devin-dumais reviewed (commented):
@shailesh1210 when I try to run this code locally (`yarn install && yarn start`) I get this error:
`Failed to compile
./src/containers/UpcomingCourses/UpcomingCourses.js
Module not found: Can't resolve 'react-calendar' in '~/apex-agility/src/containers/UpcomingCourses'`

Did you commit the updated `package.json` file that includes `react-calendar` in the dependency list to this branch?

shailesh1210 committed:
Updated package.json

devin-dumais reviewed (approved):
Great work on this 👍

shailesh1210 committed:
Merge branch 'develop' into page/upcoming-courses-2.0
